install_config/persistent_storage/pod_security_context.adoc

```diff
@@ -12,262 +12,728 @@ toc::[]
 
-SELinux options are specified as follows:
+User IDs can be defined in the container image or in the pod spec. In the pod
+definition, a single user ID can be defined globally to all containers, or
+specific to individual containers (or both). A user ID is supplied as shown in
+the pod definition fragment below:
 
+[[pod-user-id-99]]
 ====
 [source,yaml]
 ----
-apiVersion: v1
-kind: Pod
-metadata:
-  name: ebs-web
 spec:
   containers:
-    - name: web
-      image: nginx
-      ports:
-        - name: web
-          containerPort: 80
-      volumeMounts:
-          - name: ebs-volume
-            mountPath: "/usr/share/nginx/html"
-  securityContext:
-    seLinuxOptions:
-    level: "s0:c123,c456"
-  volumes:
-    - name: ebs-volume
-      awsElasticBlockStore:
-      volumeID: <VOLUME ID>
+  - name: ...
+    securityContext:
+      runAsUser: 99
 ----
 ====
 
-Currently the list of volumes which support SELinux management includes:
+ID 99 in the above is container-specific and matches the owner ID on the export.
+If the NFS export's owner ID was 54321, then that number would be used in the
+pod definition. Specifying `*securityContext*` outside of the container
+definition makes the ID global to all containers in the pod.
+
+Similar to group IDs, user IDs may be validated according to policies set in the
+SCC and/or project. If the SCC's `*runAsUser*` strategy is set to *RunAsAny*,
+then any user ID defined in the pod definition or in the image is allowed. This
+means even a UID of 0 (root) is allowed!
+
+If, instead, the `*runAsUser*` strategy is set to *MustRunAsRange*, then a
+supplied user ID will be validated against a range of allowed IDs. If the pod
+supplies no user ID, then the default ID is the minimum value of the range of
+allowable user IDs.
+
+Getting back to the earlier link:#nfs-example[NFS example], the container needs
+its UID set to 99, which is shown in the pod fragment above. Assuming the
+*default* project and the *restricted* SCC, the pod's requested user ID of 99
+will *not* be allowed, and therefore the pod will fail. The pod fails because:
+
+- it requests 99 as its user ID,
+- all available SCCs use *MustRunAsRange* for their `*runAsUser*` strategy, so UID
+range checking is required, and
+- 99 is not included in the SCC or project's user ID range.
+
+To fix this situation:
+
+- A new SCC could be created with the appropriate user ID range (recommended).
+- The *restricted* SCC could be modified to include 99 within its minimum and
+maximum user ID range (not recommended).
+- The *restricted* SCC could be modified to use *RunAsAny* for the `*runAsUser*`
+value, thus eliminating ID range checking (strongly not recommended, as
+containers could run as root).
+- The *default* project's UID range could be changed to allow a user ID of 99 (not
+generally advisable because only a single range of user IDs can be specified).
+- A new project could be created with the appropriate user ID range defined (not
+covered here).
```

Comment:
Generally, the parenthesized blurbs should either be eliminated (in the case of "not covered here") or converted to more substantial separate sentences.